symengine/cwrapper.cpp

```diff
@@ -489,6 +489,47 @@ char* ascii_art_str()
     return cc;
 }
 
+
+void ntheory_gcd(basic s, const basic a, const basic b)
+{
+    s->m = SymEngine::gcd(static_cast<const Integer &>(*(a->m)), static_cast<const Integer &>(*(b->m)));
+}
+
+void ntheory_lcm(basic s, const basic a, const basic b)
+{
+    s->m = SymEngine::lcm(static_cast<const Integer &>(*(a->m)), static_cast<const Integer &>(*(b->m)));
+}
+
+void ntheory_nextprime(basic s, const basic a)
+{
+    s->m = SymEngine::nextprime(static_cast<const Integer &>(*(a->m)));
+}
+
+void ntheory_mod(basic s, const basic n, const basic d)
+{
+    s->m = SymEngine::mod(static_cast<const Integer &>(*(n->m)), static_cast<const Integer &>(*(d->m)));
+}
+
+void ntheory_quotient(basic s, const basic n, const basic d)
+{
+    s->m = SymEngine::quotient(static_cast<const Integer &>(*(n->m)), static_cast<const Integer &>(*(d->m)));
+}
+
+void ntheory_fibonacci(basic s, unsigned long a)
+{
+    s->m = SymEngine::fibonacci(a);
+}
+
+void ntheory_lucas(basic s, unsigned long a)
+{
+    s->m = SymEngine::lucas(a);
+}
+
+void ntheory_binomial(basic s, const basic a, unsigned long b)
+{
+    s->m = SymEngine::binomial(static_cast<const Integer &>(*(a->m)), b);
```

Comment:
You should never cast to a type you are not sure of. At least a debug time check is needed before casting to a type. See, `integer_get_mpz` for an example